handler/health/health.go

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE
+
+package health
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/m3db/m3ctl/handler"
+	"github.com/m3db/m3x/instrument"
+
+	"github.com/gorilla/mux"
+)
+
+const (
+	ok          = "OK"
+	fail        = "FAIL"
+	healthURL   = "/health"
+	unknownName = "unkown"
+)
+
+type healthCheckResult struct {
+	Host         string        `json:"host"`
+	Timestamp    time.Time     `json:"timestamp"`
+	ResponseTime time.Duration `json:"response_time"`
+	Status       string        `json:"status"`
+}
+
+type controller struct {
+	iOpts instrument.Options
+}
+
+// NewController creates a new rules controller
+func NewController(iOpts instrument.Options) handler.Controller {
+	return &controller{iOpts: iOpts}
+}
+
+func healthCheck() string {
```

Comment:
This is the healthCheck in the skeleton implementation. It will be more interesting in the future.